### PR TITLE
Move from public and private keys to 1 combined keypair

### PR DESCRIPTION
**Pull request by kathrynfejer**
- Standardizes how xtt handles public and private keys by making them one structure a "keypair" which used to be called a ASN.1 wrapped private key. 
- Adds functions to read and write a keypair to a file. 
- Updates the tool to pass the keypair instead of both the public and private keys.
- Reorganizes util directory to remove extraneous functions and files

### CODE REVIEW

File: tool/parse_cli.c [hunk 1]
```diff
@@ -110,79 +98,35 @@ void parse_genx509cert_cli(int argc, char **argv, struct cli_params *params)
...
-        }
-    }
-}
-
-
 static
 void parse_genroot_cli(int argc, char **argv, struct cli_params *params)
 {
-    params->pubkey = "root_pub.bin";
+    params->keypair = "root_keypair.bin";
```
Comment (zanebeckwith) [inline tool/parse_cli.c]: Can we change this default to something like `root_keys.asn1.bin`? Then, all the other places this is used will have to be changed...

File: tool/parse_cli.c [hunk 2]
```diff
@@ -65,36 +59,30 @@ void parse_genx509cert_cli(int argc, char **argv, struct cli_params *params)
...
-    params->pubkey = "pub.bin";
+    params->keypair = "keypair.bin";
 
     const char *usage_str = "Generate x509 certificate.\n\n"
         "Usage: %s %s [-h] [-v <file>] [-b <file>] [-d <file>] [-c <file>]\n"
         "\tOptions:\n"
         "\t\t-h --help             Display this message.\n"
-        "\t\t-v --privkey          Private key input location [default = priv.bin]\n"
-        "\t\t-b --pubkey           Public key input location [default = pub.bin]\n"
+        "\t\t-k --keypair          Key pair input location [default = keypair.bin]\n"
```
Comment (zanebeckwith) [inline tool/parse_cli.c]: Let's match this to whatever we make the default for `genkeypair`

File: tool/parse_cli.c [hunk 3]
```diff
@@ -202,50 +146,41 @@ static
...
     const char *usage_str = "Generate server's certificate.\n\n"
         "Usage: %s %s [-h] [-r <file>] [-p <file>] [-v <file>] [-b <file>] [-x <file>] [-c <file>]\n"
         "\tOptions:\n"
         "\t\t-h --help                    Display this message.\n"
         "\t\t-r --rcert                   Root certificate input location [default = root_cert.bin]\n"
-        "\t\t-p --rpriv                   Root private key input location [default = root_priv.bin]\n"
-        "\t\t-v --serverpriv              Server private key input location [default = server_priv.bin]\n"
-        "\t\t-b --serverpub               Server public key input location [default = server_pub.bin]\n"
+        "\t\t-k --rkeypair                Root key pair input location [default = root_keypair.bin]\n"
+        "\t\t-s --skeypair                Server key pair input location [default = server_keypair.bin]\n"
```
Comment (zanebeckwith) [inline tool/parse_cli.c]: Can we make this default `server_keys.asn1.bin`? And, again, we'll have to propagate this to all the other places that use this.

**Comment by zanebeckwith** [comment]
I just pushed a commit updating the README to reflect the changes you made regarding the defaults.

Looks good to me! Go ahead and rebase it, and I'll merge it!

Thanks!

**Comment by kathrynfejer** [comment]
Only took me six months, but I think I somewhat figured out this rebasing thing. It should be ready now!!

**Comment by zanebeckwith** [comment]
Awesome!

(I hope git rebasing hasn't been the only thing you've learned during your time at Xaptum...)

**Pull request merged**